automata/fa/dfa.py

```diff
@@ -318,15 +318,9 @@ def _merge_states(self, retain_names=False):
                         else:
                             processing.add(XdiffY)
 
-        def get_name(eq, i):
-            if retain_names:
-                return list(eq)[0] if len(eq) == 1 else DFA._stringify_states(eq)
-
-            return str(i)
-
         # now eq_classes are good to go, make them a list for ordering
         eq_class_name_pairs = [
-            (eq, get_name(eq, i))
+            (eq, frozenset(eq) if retain_names else str(i))
```

Comment:
@eliotwrobson Not trying to [pre-optimize](https://stackify.com/premature-optimization-evil/) here 😅, but do you think it would be more performant for the `retain_names` check to be _outside_ the loop so it only runs once?

```python
if retain_names:
    eq_class_name_pairs = [
        (eq, frozenset(eq))
        for eq in eq_classes
    ]
else:
    eq_class_name_pairs = [
        (eq, str(i))
        for i, eq in enumerate(eq_classes)
    ]
```

Reply:
Probably doesn't make a huge difference but I'll go ahead and rewrite this line.

Reply:
@caleb531 did a mini-rewrite and looks cleaner now!